app/views/LocationTracking.js

```diff
@@ -36,30 +42,91 @@ class LocationTracking extends Component {
     super(props);
 
     this.state = {
+      timer_intersect: null,
       isLogging: '',
     };
   }
 
   componentDidMount() {
     BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
     GetStoreData('PARTICIPATE')
-      .then(isParticipating => {
-        console.log(isParticipating);
-
-        if (isParticipating === 'true') {
-          this.setState({
-            isLogging: true,
-          });
-          this.willParticipate();
-        } else {
-          this.setState({
-            isLogging: false,
-          });
-        }
-      })
-      .catch(error => console.log(error));
+        .then(isParticipating => {
+          if (isParticipating === 'true') {
+            this.setState({
+              isLogging: true,
+            });
+            this.willParticipate();
+          } else {
+            this.setState({
+              isLogging: false,
+            });
+          }
+        })
+        .catch(error => console.log(error));
+
+    let timer_intersect = setInterval(this.intersect_tick, 1000 * 60 * 60 * 12); // once every 12 hours
+    // DEBUG:  1000 * 10); // once every 10 seconds
+
+    this.setState({
+      timer_intersect,
+    });
   }
+
+  intersect_tick = () => {
+    // This function is called once every 12 hours.  It should do several things:
```

Comment:
* If the function needs an outline/table of contents, please split it into smaller functions
* Use async await

```js
const authorities = await getUserAuthorities();
if (authority_list) {
  await fetchAuthorities(authority_list);
}
```

Reply:
I agree, but this wasn't introduced by the PR.  Not really this contributor's responsibility.

Reply:
This is not changed by me. May be an issue with some space.